src/test/java/net/masterthought/cucumber/util/StepNameFormatterTest.java

```diff
@@ -0,0 +1,94 @@
+package net.masterthought.cucumber.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import net.masterthought.cucumber.generators.integrations.PageTest;
+import net.masterthought.cucumber.json.MatchArgument;
+import net.masterthought.cucumber.json.Step;
+import org.junit.Before;
+import org.junit.Test;
+
+public class StepNameFormatterTest extends PageTest {
+    @Before
+    public void setUp() {
+        setUpWithJson(SAMPLE_JSON);
```

Comment:
To test without JSON context it would be necessary to create Step and other objects manually. But they have no constructors or setters (and the attributes are final), so I used the JSON context to have the objects created.